app/src/main/java/com/infomaniak/swisstransfer/ui/MainApplication.kt

```diff
@@ -65,4 +73,17 @@ class MainApplication : Application(), Configuration.Provider {
             }
         }
     }
+
+    private suspend fun collectTransfers() {
+        transferManager.getTransfers(TransferDirection.SENT).collect(::deleteExpiredTransfers)
+        transferManager.getTransfers(TransferDirection.RECEIVED).collect(::deleteExpiredTransfers)
```

Comment:
Deleting expired transfers might be on KMP ? That way, we avoid doing the work on both platform ?

Reply:
Yes totally, the logic to detect an old expired Transfer could be shared.

Reply:
It's been moved.